source/grid/grid_out.cc

```diff
@@ -3051,19 +3051,24 @@ GridOut::write_vtk(const Triangulation<dim, spacedim> &tria,
   const auto co_faces = relevant_co_faces(tria);
 
   // Write cells preamble
-  const int n_cells = tria.n_active_cells() + faces.size() + co_faces.size();
+  const int n_cells = vtk_flags.draw_cells * tria.n_active_cells() +
```

Comment:
This and the next two lines make use of the implicit conversion of `bool` to `int`. That's hard to read. Make it explicit:
```suggestion
  const int n_cells = (vtk_flags.draw_cells ? tria.n_active_cells() : 0 ) +
```
and similarly for the next couple of lines